app/WebParsing/UtsgJsonParser.hs

```diff
@@ -91,14 +103,26 @@ meetingUpdates m = [ MeetingCode =. meetingCode m
                    , MeetingExtra =. meetingExtra m
                    ]
 
-newtype DB = DB { dbData :: (Courses, [MeetTime]) }
+newtype DB = DB { dbData :: [MeetTime]}
+  deriving Show
+
+newtype DBList = DBList [DB]
   deriving Show
 
-instance FromJSON DB where
-    parseJSON (Object o) = do
-      course <- parseJSON (Object o)
-      session :: T.Text <- o .:? "section" .!= "F"
-      meetingTimesMap :: HM.HashMap T.Text MeetTime <- o .:? "meetings" .!= HM.empty
-      let allMeetingsTimes = map (\m -> m {meetInfo = (meetInfo m) { meetingCode = coursesCode course, meetingSession = session}}) (HM.elems meetingTimesMap)
-      return $ DB (course, allMeetingsTimes)
-    parseJSON _ = fail "Invalid section"
+instance FromJSON DBList where
+  parseJSON = withObject "Expected an Object for DBList" $ \o -> do
+    maybeOrg <- o .:? "payload"
+    case maybeOrg of
+      Just payload -> do
+        pageableCourse <- payload .: "pageableCourse"
+        courses <- pageableCourse .: "courses"
+        dbList <- mapM parseCourse courses
+        return $ DBList dbList
+        where
+          parseCourse courseObj = do
```

Comment:
This should really still be in the typeclass definition for `FromJSON DB`, so that you can use `mapM parseJSON courses` or something similar.